### DISCUSSION TITLE
Why aren't my scheduled jobs running?

### DISCUSSION THREAD

**Discussion by Ayo-Awe** [riverqueue/river#580] [Q&A]
I'm trying to schedule jobs for a later date with riverqueue, but it doesn't seem to be working. If I take off the ScheduledAt insert option everything seems to work fine otherwise, it doesn't work.

Here's my setup
![F4DCDA08-4440-4253-9BA2-AFFC444EFB42](https://github.com/user-attachments/assets/f06060d4-382d-4d5a-ab02-2041426cce34)

![image](https://github.com/user-attachments/assets/58ed6ab5-9bb8-44c1-837a-7e36debb4211)

![image](https://github.com/user-attachments/assets/dceb6e50-a9d0-494e-90cb-cc158d83a8ec)

**Comment by bgentry** [comment]
Hmm, have you looked at the actual job row being produced by this insert? For a working example, I would expect its state to be `scheduled` and its `scheduled_at` to be the time in the future corresponding to your delivery date.

Normally we see the scheduled time set [via the insert opts as part of the insert function call](https://riverqueue.com/docs/scheduled-jobs), but I believe the method you’re doing (pulling it from args via `InsertOpts` method on args) _should_ also work.

**Comment by Ayo-Awe** [comment]
I've checked the job row, and it seems like the job isn't being attempt at all.

I scheduled the job for a minute later, but it hasn't run at all. I've even tried larger intervals but to no avail

![image](https://github.com/user-attachments/assets/7c337f86-0095-4ee7-9205-132690038bc8)

**Comment by brandur** [comment]
@Ayo-Awe One thing I don't see in any of your sample code is a call to `Start` on the client. Make sure you to start it, or it won't work jobs or run maintenance services like the job scheduler.

**Comment by Ayo-Awe** [comment]
My bad, I actually called start but I forgot to include it in the sample code
<img width="865" alt="Screenshot 2024-09-07 at 10 25 50" src="https://github.com/user-attachments/assets/7bed24cb-4e4a-4b0c-97da-28f8b6f49872">

The weird part is that it's only scheduled tasks that don't run, regular tasks run fine

**Comment by brandur** [comment]
@Ayo-Awe Try running River with a debug-level logger. You should be able to look for log lines that start with "JobScheduler" which should give a hint as to what's going on.

**Comment by Ayo-Awe** [comment]
@brandur I just checked the debug logs and I can't find anything on the scheduler. 
However, there are some logs on unsuccessful leadership bid

<img width="1000" alt="Screenshot 2024-09-08 at 09 56 09" src="https://github.com/user-attachments/assets/f8ccc24a-686d-4095-b58d-810401392496">

**Reply by bgentry** [reply]
Got anything in your `river_leader` table? Also are you fully caught up on migrations?

Is there only one node running in the environment these logs are from?
